### PR TITLE
"trackPublicationFailed" errors related to Track Names

### PR DESCRIPTION
**Pull request by manjeshbhargav**
@markandrus This PR updates `twilio-video-errors.js` with additional `TwilioError`s, and adds tests to ensure that attempts to publish Tracks with either duplicate names or with names which have more than 128 characters.

### CODE REVIEW

File: lib/util/twilio-video-errors.js [hunk 1]
```diff
@@ -38,6 +38,108 @@ exports.AccessTokenInvalidError = AccessTokenInvalidError;
...
+  TwilioError.call(this,
+    20107,
+    'Invalid Access Token signature'
+  );
+}
+inherits(AccessTokenSignatureInvalidError, TwilioError);
+exports.AccessTokenSignatureInvalidError = AccessTokenSignatureInvalidError;
+Object.defineProperty(TwilioErrorByCode, 20107, { value: AccessTokenSignatureInvalidError });
+
+/**
```
Comment (markandrus) [inline lib/util/twilio-video-errors.js]: Funny, I could've sworn we had these already...

File: lib/util/twilio-video-errors.js [hunk 2]
```diff
@@ -242,6 +344,142 @@ exports.RoomNotFoundError = RoomNotFoundError;
...
+  TwilioError.call(this,
+    53114,
+    'Room creation parameter(s) incompatible with the Room type'
+  );
+}
+inherits(RoomInvalidParametersError, TwilioError);
+exports.RoomInvalidParametersError = RoomInvalidParametersError;
+Object.defineProperty(TwilioErrorByCode, 53114, { value: RoomInvalidParametersError });
+
+/**
```
Comment (markandrus) [inline lib/util/twilio-video-errors.js]: Hmm, these should never be raised at the client if I am not mistaken. Maybe we can skip these? We may need a way to identify which errors can and should not be raised at the Client...

File: lib/util/twilio-video-errors.js [hunk 3]
```diff
@@ -327,6 +565,23 @@ exports.ParticipantNotFoundError = ParticipantNotFoundError;
...
+  TwilioError.call(this,
+    53205,
+    'Participant disconnected because of duplicate identity'
+  );
+}
+inherits(ParticipantDuplicateIdentityError, TwilioError);
+exports.ParticipantDuplicateIdentityError = ParticipantDuplicateIdentityError;
+Object.defineProperty(TwilioErrorByCode, 53205, { value: ParticipantDuplicateIdentityError });
+
+/**
```
Comment (markandrus) [inline lib/util/twilio-video-errors.js]: +1, this one is useful

File: lib/util/twilio-video-errors.js [hunk 4]
```diff
@@ -395,6 +650,23 @@ exports.TrackNameCharsInvalidError = TrackNameCharsInvalidError;
...
+  TwilioError.call(this,
+    53304,
+    'Track name is duplicated'
+  );
+}
+inherits(TrackNameIsDuplicatedError, TwilioError);
+exports.TrackNameIsDuplicatedError = TrackNameIsDuplicatedError;
+Object.defineProperty(TwilioErrorByCode, 53304, { value: TrackNameIsDuplicatedError });
+
+/**
```
Comment (markandrus) [inline lib/util/twilio-video-errors.js]: +1, also useful

File: lib/util/twilio-video-errors.js [hunk 5]
```diff
@@ -512,3 +801,5 @@ function ConfigurationAcquireTurnFailedError() {
 inherits(ConfigurationAcquireTurnFailedError, TwilioError);
 exports.ConfigurationAcquireTurnFailedError = ConfigurationAcquireTurnFailedError;
 Object.defineProperty(TwilioErrorByCode, 53501, { value: ConfigurationAcquireTurnFailedError });
+
+
```
Comment (markandrus) [inline lib/util/twilio-video-errors.js]: I know this is auto-gen, but can we remove these two lines?

File: test/integration/spec/connect.js [hunk 6]
```diff
@@ -20,6 +20,7 @@ const LocalAudioTrack = require('../../../lib/media/track/localaudiotrack');
 const LocalDataTrack = require('../../../lib/media/track/localdatatrack');
 const LocalVideoTrack = require('../../../lib/media/track/localvideotrack');
 const TwilioError = require('../../../lib/util/twilioerror');
+const { TrackNameIsDuplicatedError, TrackNameTooLongError } = require('../../../lib/util/twilio-video-errors');
```
Comment (markandrus) [inline test/integration/spec/connect.js]: Nice to be able to import the exact errors and test for them 👍

File: test/integration/spec/localparticipant.js [hunk 7]
```diff
@@ -364,6 +365,63 @@ const isSafari = guess === 'safari';
...
+                const { name } = tracks.find(track => track.kind === 'audio');
+                console.log('Name:', name);
+                return createLocalAudioTrack({ name });
+              },
+              scenario: 'a LocalTrack whose name is the same as one of the currently published LocalTracks',
+              TwilioError: TrackNameIsDuplicatedError
+            },
+            {
+              createLocalTrack() {
+                const name = '0'.repeat(130);
```
Comment (markandrus) [inline test/integration/spec/localparticipant.js]: Cool, didn't know about `repeat`. So the max length is what? 129?

**Comment by markandrus** [comment]
A few comments, but mostly LGTM. Let's cherry-pick onto other branch when it's green.

File: test/integration/spec/localparticipant.js (hunk 7 above)
Comment (manjeshbhargav) [reply test/integration/spec/localparticipant.js]: @markandrus 128 characters. I could change the above number to 129.

File: lib/util/twilio-video-errors.js (hunk 2 above)
Comment (manjeshbhargav) [reply lib/util/twilio-video-errors.js]: @markandrus On second thought, I think we should hold off on removing certain errors manually until we figure out a way to generate only client specific errors. I'm sure there are other errors apart from this that we might have to remove.

**Comment by markandrus** [comment]
Closing--this has been cherry-picked onto the other PR.

**Pull request closed without merging**